File: MOC Develop SQL 2005/Labfiles/Lab07/Solution/vb/ManagedUtilities1/SaveXML.vb
Imports System
Imports System.Data
Imports System.Data.Sql
Imports System.Data.SqlServer
Imports System.Data.SqlTypes
Imports System.Xml


Partial Public Class StoredProcedures
    <SqlProcedure(Name:="SaveXML")> _
Public Shared Sub SaveXML(ByVal XmlData As SqlXml, _
        ByVal FileName As SqlString)

        Dim xmlDoc As New XmlDocument()
        xmlDoc.LoadXml(XmlData.Value)
        xmlDoc.Save(FileName.Value)

    End Sub
End Class

File: MOC Develop SQL 2005/Labfiles/Lab07/Solution/cs/ManagedUtilities1/SaveXML.cs
using System;
using System.Data;
using System.Data.Sql;
using System.Data.SqlServer;
using System.Data.SqlTypes;
using System.Xml;

namespace FileUtilities
{
	public partial class StoredProcedures
	{
		[SqlProcedure(Name = "SaveXML")]
		public static void SaveXML(SqlXml XmlData,
							   SqlString FileName)
		{
			XmlDocument xmlDoc = new XmlDocument();
			xmlDoc.LoadXml(XmlData.Value);
			xmlDoc.Save(FileName.Value);
		}
	};
}

